pulsar-client-cpp/tests/ProducerTest.cc

```diff
@@ -210,3 +213,84 @@ TEST(ProducerTest, testBacklogQuotasExceeded) {
 
     client.close();
 }
+
+TEST(ProducerTest, testMaxMessageSize) {
+    Client client(serviceUrl);
+
+    const std::string topic = "ProducerTest-MaxMessageSize-" + std::to_string(time(nullptr));
+
+    Consumer consumer;
+    ASSERT_EQ(ResultOk, client.subscribe(topic, "sub", consumer));
+
+    Producer producer;
+    ASSERT_EQ(ResultOk, client.createProducer(topic, producer));
+
+    std::string msg = std::string(maxMessageSize / 2, 'a');
+    ASSERT_EQ(ResultOk, producer.send(MessageBuilder().setContent(msg).build()));
+    Message message;
+    ASSERT_EQ(ResultOk, consumer.receive(message));
+    ASSERT_EQ(msg, message.getDataAsString());
+
+    std::string orderKey = std::string(maxMessageSize, 'a');
+    ASSERT_EQ(ResultMessageTooBig, producer.send(MessageBuilder().setOrderingKey(orderKey).build()));
+
+    ASSERT_EQ(ResultMessageTooBig,
+              producer.send(MessageBuilder().setContent(std::string(maxMessageSize, 'b')).build()));
+
+    client.close();
+}
+
+TEST(ProducerTest, testNoBatchMaxMessageSize) {
```

Comment:
It's nearly the same with `testMaxMessageSize` except the difference of the producer configuration. You can simplify the test via `TEST_P`. See `MessageChunkingTest` for example. BTW, I think these tests could also be moved to `MessageChunkingTest`.